src/compiletest.md

```diff
@@ -0,0 +1,183 @@
+# `compiletest`
+## Introduction
+`compiletest` is the main test harness of the Rust test suite.  It allows test authors to organize large numbers of tests (the
+Rust compiler has many thousands), efficient test execution (parallel execution is supported), and allows the test author to
+configure behavior and expected results of both individual and groups of tests.
+
+`compiletest` tests may check test code for success, for failure or in some cases, even failure to compile.  Tests are
+typically organized as a Rust source file with annotations in comments before and/or within the test code, which serve to
+direct `compiletest` on if or how to run the test, what behavior to expect, and more.
+
+The tests themselves are typically (but not always) organized into "suites"--for example, `run-pass`, a folder
+representing tests that should succeed, `run-fail`, a folder holding tests that should compile successfully, but return
+a failure (non-zero status), `compile-fail`, a folder holding tests that should fail to compile, and many more.  The various
+suites are defined in `src/tools/compiletest/src/common.rs` in the `pub struct Config` declaration.  And a very good 
+introduction to the different suites of compiler tests along with details about them can be found
+[here, at Brian Anderson's blog](https://brson.github.io/2017/07/10/how-rust-is-tested#s-ct).
+
+## Adding a new test file
+Simply create your new test in the appropriate location under `/src/test`.  No registration of test files is necessary as 
+`compiletest` will scan the `src/test` subfolder recursively, and will execute any Rust source files it finds as tests.
+
+## Header Commands
+Source file annotations which appear in comments near the top of the source file *before* any test code are known as header
+commands.  These commands can instruct `compiletest` to ignore this test, set expectations on whether it is expected to 
+succeed at compiling, or what the test's return code is expected to be.  Header commands (and their inline counterparts,
+Error Info commands) are described more fully 
+[here](https://github.com/rust-lang/rust/blob/master/src/test/COMPILER_TESTS.md).
```

Comment:
I think this can be a link to the chapter from #47

Reply:
Do you want a link in the doc to the PR or to wait until it lands and can be linked to under `master`?

Reply:
> to wait until it lands and can be linked to under master

I think this would be best, unless you are under a time constraint.

Reply:
I think you missed this one